packages/hardhat-core/test/internal/core/config/config-resolution.ts

```diff
@@ -645,6 +647,60 @@ describe("Config resolution", () => {
         assert.deepEqual(config.networks.hardhat, {
           ...networkConfig,
           minGasPrice: new BN(10),
+          chains: defaultHardhatNetworkParams.chains,
+        });
+      });
+
+      describe("chains", function () {
+        it("should default to default", function () {
+          const resolvedConfig = resolveConfig(__filename, {});
+          assert.deepEqual(
+            Array.from(resolvedConfig.networks.hardhat.chains.entries()),
+            Array.from(defaultHardhatNetworkParams.chains.entries())
+          );
+        });
+        describe("mixing defaults and user configs", function () {
+          const userConfig = {
+            networks: {
+              hardhat: { chains: { 1: { hardforkHistory: { london: 999 } } } },
+            },
+          };
+          const resolvedConfig = resolveConfig(__filename, userConfig);
+          it("If the user provides values for a chain that's included in the default, should use the users' values, and ignore the defaults for that chain.", function () {
+            assert.deepEqual(resolvedConfig.networks.hardhat.chains.get(1), {
+              hardforkHistory: new Map([[HardforkName.LONDON, 999]]),
+            });
+            assert.deepEqual(resolvedConfig.networks.hardhat.chains.get(1), {
```

Comment:
This second assert is not necessary, right?

Reply:
Removed in https://github.com/nomiclabs/hardhat/pull/1928/commits/540c97ffeb3f16ca75a53c559bad66608f6a4402